notion_integration/notion_processing.py

```diff
@@ -1,120 +1,120 @@
+                    {
+                        "type": "text",
+                        "text": {"content": dict_thought["highlight"]},
+                    }
+                ]
+            },
+            "Date": {"date": {"start": dict_thought["date"].replace(".", "-")}},
+            "ToCorrect": {"checkbox": False},
+            "Location": {"number": int(dict_thought["location"])},
+            "Book": {
+                "rich_text": [
+                    {"type": "text", "text": {"content": dict_thought["title"]}}
+                ]
+            },
+            "Tags": {"select": {"name": "Understanding"}},
+            "Idx": {"number": id_number},
+        },
     }
 
     return data
 
+
 def send_thoughts_to_database(data):
     """This function sends the data to the notion database"""
     notion_book_databse = NotionDatabase()
     last_idx = notion_book_databse.retrive_last_idx_number()
 
     for idx, item in tqdm(enumerate(data)):
-        idx=idx+last_idx # 592 TODO: make it more robust now its assuming its always first row with latest number
-        data = prepare_data_for_notion(idx+1, item, notion_book_databse.notion_database_id)
-        notion_book_databse.send_to_notion(data)
+        idx = (
+            idx + last_idx
+        )  # 592 TODO: make it more robust now its assuming its always first row with latest number
+        data = prepare_data_for_notion(
+            idx + 1, item, notion_book_databse.notion_database_id
+        )
+        notion_book_databse.send_data(data)
 
-    logging.info(f"{idx-last_idx+1} rows sent to Notion. New last idx {idx+1}")
+    logging.info(
+        f"{len(data)-last_idx+1} rows sent to Notion. New last idx {len(data)+1}"
+    )
 
 
-def main(filepath):
-    """This function is used to read parsed data from Readwise format and send to the notion database"""
+def main(csv_filepath):
+    """This function is used to read parsed data from Readwise format and
+    send to the notion database"""
     # filepath = 'data_output\Fix-Zero-To-One.csv'
-    data = read_csv_rows_into_dict(filepath)
-    logging.info(f"Read {len(data)} rows from {filepath}")
+    data = read_csv_rows_into_dict(csv_filepath)
+    logging.info(f"Read {len(data)} rows from {csv_filepath}")
     send_thoughts_to_database(data)
 
 
 def read_input_file_path():
-    parser = argparse.ArgumentParser(description='Readwise to Notion')
-    parser.add_argument('-f', '--filepath', type=str, metavar='', required=True, help='Filepath to the csv file')
+    """This function is used to read the input file path from the command line"""
+    parser = argparse.ArgumentParser(description="Readwise to Notion")
+    parser.add_argument(
+        "-f",
+        "--filepath",
+        type=str,
+        metavar="",
+        required=True,
+        help="Filepath to the csv file",
+    )
     args = parser.parse_args()
     return args.filepath
 
+
 if __name__ == "__main__":
     filepath = read_input_file_path()
```

Comment:
Ja unikam pisania czekoglwiek pod if == 'main' i polecam też unikać. 
wszystkie zmienne napisane pod tym są globalne i są dostępne w innych funkcjach a nawet mozne je improtwac do innych plikow. Przez przypadek można użyć tych zmiennych funkcjach, szczególnie jak mają podobne nazwy. np. func(file_path: str): i odwołasz sie do filepath i kod będzie działać, a czy dobrze to zależy...

Reply:
o to tego nie wiedziałem że to jest widoczne w cały scope. Zawsze tak args przekazywałem. Myślałem że to good practice :)

Reply:
done added args so there is not variable name to be mixed